src/Symfony/Component/Yaml/Inline.php

```diff
@@ -506,7 +506,12 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = []
 
             if ('!php/const' === $key) {
                 $key .= self::parseScalar($mapping, $flags, [':', ' '], $i, false, [], true);
-                $key = self::evaluateScalar($key, $flags);
+                if ('!php/const:' === $key && ':' !== $mapping[$i]) {
```

Comment:
This is a bug fix. `{!php/const: foo}` is currently interpreted as legacy `php/const:` tag + `''` as the const  value while it should end up as `['' => 'foo']`.

Reply:
Note that this must be removed once merged to 4.3.

Reply:
Ref https://github.com/symfony/symfony/pull/35318 btw